File: 0651-4-keys-keyboard/0651-4-keys-keyboard.java
class Solution {
    public int maxA(int n) {
        int[] dp = new int[n+1];
        for(int i = 1; i <= n; ++i) {
            dp[i] = i;
            for(int j = 1; j < i-2; ++j) {
                int total = dp[j] + dp[j]*(i-j-2);
                dp[i] = Math.max(dp[i], total);
            }
        }
        return dp[n];
    }
}

File: 0651-4-keys-keyboard/0651-4-keys-keyboard.js
/**
 * @param {number} n
 * @return {number}
 */
var maxA = function(n) {
    let dp = new Array(n+1).fill(0);
    for(let i = 1; i <= n; ++i) {
        dp[i] = i;
        for(let j = 1; j < i - 2; ++j) {
            const total = dp[j] + dp[j] * (i-j-2);
            dp[i] = Math.max(dp[i], total);
        }
    }
    return dp[n];
};